Add Image.save_exr()

Zylann opened the pull request:
Added a function to complement `save_png()`, because 16-bit and 32 bit float formats can't be saved from Godot.
This notably allows to save terrain data as a proper image format and eases interoperability with other software (although if we want terrain EXR to be efficient for use in Godot, we'll have to make it support single-channel properly).

The following formats can be saved:
```
RH
RGH
RGBH
RGBAH
RF
RGF
RGBF
RGBAF
R8
RG8
RGB8
RGBA8
```

All tested with this project:
[TestExrSave.zip](https://github.com/godotengine/godot/files/3463185/TestExrSave.zip)
Results come up correct when checked in Gimp and RenderDoc.

Closes https://github.com/godotengine/godot/issues/17450

Zylann committed:
Add Image.save_exr()

fire commented on modules/tinyexr/image_saver_tinyexr.cpp:
```diff
@@ -0,0 +1,179 @@
+}
+
+static int get_pixel_type(Image::Format p_format) {
+	switch (p_format) {
```
Do we want to support standard rgba8?

Zylann replied:
I was wondering about this. The last type Tinyexr supports is `TINYEXR_PIXEL_TYPE_UINT`, but where is it specified that it would be 8-bit?

fire replied:
I think openexr only supports 16bit and 32 bit floats, but you can save a regular 8bit int as a value from 0 to 1 directly.

Zylann replied:
I guess I could use 16-bit float for these

fire replied:
I would prefer to use the default 16bit openexr format for int8 colors.

akien-mga commented on modules/tinyexr/image_saver_tinyexr.h:
```diff
@@ -0,0 +1,8 @@
+#ifndef IMAGE_SAVER_TINYEXR_H
```
You should add the Godot license header here and in the cpp.
Note that line 2 should include the filename, and that is has a trailing `*/` that should be kept properly aligned.

akien-mga commented:
Could you add the relevant documentation for the new method? It should likely mention that the method returns `[constant ERR_UNAVAILABLE]` if the tinyexr module is disabled.

Zylann commented:
- Added license header
- Added `grayscale` parameter
- Reversed channels as TinyEXR examples recommend
- Support 8-bit pixel types by converting them to half
- Added documentation

Updated test project:
[TestExrSave.zip](https://github.com/godotengine/godot/files/3478733/TestExrSave.zip)

akien-mga commented:
Thanks!
